lib/qbwc/active_record/job.rb

```diff
@@ -18,7 +20,12 @@ def self.add_job(name, enabled, company, worker_class)
     ar_job.request_index = 0
     ar_job.worker_class = worker_class
     ar_job.save!
-    self.new(name, enabled, company, worker_class)
+
+    jb = self.new(name, enabled, company, worker_class, requests)
+    jb.requests = requests.is_a?(Array) ? requests : [requests] unless requests.nil?
+    jb.requests_provided_when_job_added = (! requests.nil?)
```

Comment:
I don't think so; `requests_provided_when_job_added` should only be assigned when invoked via `add_job`; counter-intuitively, the initializer is invoked multiple times during a session as a job is processed.